### PR TITLE
Feature/human sort

### PR DESCRIPTION
**Pull request by tinytyranid**
This PR adds sorting of human chromosomes so that mitochondrial and non standard chromosomes comes after chrX/X and chrY/Y.

### CODE REVIEW

File: internal/bed/sorting.go [hunk 1]
```diff
@@ -88,3 +109,42 @@ func naturalStringCompare(a, b string) int {
...
+	}
+	if bIdx != -1 && aIdx == -1 {
+		return 1
+	}
+	return naturalStringCompare(a, b)
+}
+
+// Returns position if item is in slice and
+// -1 if item is not in slice
+func idxInSlice(slice []string, item string) int {
```
Comment (marrip) [inline internal/bed/sorting.go:143]: If you made the slice a map containing the names as keys and the indices as values, you don't need to do the ranging. That would probably increase speed when sorting long bed files. You need to be aware that missing keys will return zero (default value for int) so maybe start the indices at 1. Another way would be to check if the key exists and simply handle the case where it doesn't.

File: internal/bed/sorting.go [hunk 2]
```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/maruel/natural"
 )
 
+// Note that only lowercase is used in this slice
+var humanChrOrder = []string{"1", "chr1", "2", "chr2", "3", "chr3", "4", "chr4", "5", "chr5", "6", "chr6", "7", "chr7", "8", "chr8", "9", "chr9", "10", "chr10", "11", "chr11", "12", "chr12", "13", "chr13", "14", "chr14", "15", "chr15", "16", "chr16", "17", "chr17", "18", "chr18", "19", "chr19", "20", "chr20", "21", "chr21", "x", "chrx", "y", "chry", "mt", "chrmt"}
```
Comment (marrip) [inline internal/bed/sorting.go:13]: I think if the prefix `chr` is used, the mitochondrial genome is `chrM`. Only without prefix it is `MT`.

I was also thinking (just an idea, not sure if it is feasible) if natural sorting can be applied and by using a regex for the mitochondrial genome you could do the shift to the end of the main chromosomes.

**Review by marrip — COMMENTED** [review]
Very nice to have this option - even thought this is hard to do programmatically. I had some ideas/suggestions, let me know what you think ☺️

File: internal/bed/sorting.go (hunk 2 above)
Comment (tinytyranid) [reply internal/bed/sorting.go:13]: Thank you, I was not aware about chrM this!
I see now that both chrM and chrMT is used, so I'm adding both M and chrM to the list.

File: internal/bed/sorting.go (hunk 1 above)
Comment (tinytyranid) [reply internal/bed/sorting.go:143]: This made me think a bit. 
What if we let the user provide a custom chromosome ordering as a list and then we transform it into a map for sorting? We can use the human sorting as default and then people working with non-human organisms, can also get the chromosome ordering they want?

Comment (marrip) [reply internal/bed/sorting.go:143]: I think that is actually a great idea! 👍

File: internal/bed/bed.go [hunk 3]
```diff
@@ -51,9 +57,26 @@ func (bf *Bedfile) VerifyAndHandle() error {
...
+	}
 	bf.Input = filepath.Clean(bf.Input)
 	if bf.Output != "" {
 		bf.Output = filepath.Clean(bf.Output)
 	}
 	return nil
 }
+
+// Convert provided chromosome order to map
+func chrOrderToMap(chrOrder []string) map[string]int {
```
Comment (marrip) [inline internal/bed/bed.go:76]: amazing! I love that it is customizable now 👍

**Review by marrip — APPROVED** [review]
I think it looks very good! Great work! 🎉

**Pull request merged**